drivers/dac/ltc2688/ltc2688.c

```diff
@@ -387,6 +387,8 @@ int32_t ltc2688_set_voltage(struct ltc2688_dev *dev, uint8_t channel,
 	if(code > 0xFFFF)
 		code = 0xFFFF;
 
+	dev->dac_code[channel] = code;
```

Comment:
what is the purpose of this change ? I expect the code to make it somehow to the registers of the chip but the code now simply stores it...

Reply:
To let the user know of what the actual code that is being written into the device register after taking gain and offset into consideration.

Reply:
To let the user know of what the actual code that is being written into the device register after taking gain and offset into consideration.

Reply:
It seems to me then that this is a debug feature. I would advise against implementing it this way and suggest using `pr_debug` from this file to achieve what you want.
https://github.com/analogdevicesinc/no-OS/blob/master/include/no_os_print_log.h

This computed code is information that the driver doesn't need to store in the device descriptor (proof: it's not used elsewhere than the function it's computed in: ltc2688_set_voltage).

Reply:
It's not about just printing the value, it about getting the dac codes back to application level. I believe the device descriptor should provide information about what's the current configuration of DAC registers. In this [drivers ](https://github.com/analogdevicesinc/no-OS/blob/master/drivers/dac/ad5770r/ad5770r.h#L317) as well, the dac value is being stored in the descriptor itself and it is not being used anywhere in the drivers except for setting the dac voltage

If _ltc2688_spi_read() function were not declared static, it would have been easier to read contents of the dac registers directly.
Or else I can create a global variable to store these values in the drivers itself.

Reply:
It's not about just printing the value; it's about getting the DAC codes back to the application level. I believe the device descriptor should provide information about the current configuration of DAC registers. In this [drivers ](https://github.com/analogdevicesinc/no-OS/blob/master/drivers/dac/ad5770r/ad5770r.h#L317) as well, the DAC value is being stored in the descriptor itself, and it is not being used anywhere in the drivers except for setting the DAC voltage 

If the _ltc2688_spi_read() function were not declared static, it would have been easier to read the contents of the DAC registers directly. Or else I can create a global variable to store these values in the drivers themselves.

Reply:
Ok, fine for me then if you say this should work as a cached version of the value one would read using _ltc2688_spi_read()